tools/packaging/common/envoy_bootstrap_v2.json

```diff
@@ -297,7 +297,10 @@
             ],
             "validation_context": {
               "trusted_ca": {
-                {{ if eq .pilot_cert_provider "kubernetes" }}
+                {{ if .output_key_cert_to_dir }}
+                {{ $combine := (printf "%s%s" .output_key_cert_to_dir "/root-cert.pem")}}
```

Comment:
Isn't OutputKeyCertToDir meant for debugging/alternative usage of the certs like prometheus?? Why are we using it ourselves? 

@lei-tang @costinm

Reply:
we used it internally as well to see whether we have provided the cert by ourself:  
`sa := istio_agent.NewSDSAgent(proxyConfig.DiscoveryAddress, controlPlaneAuthEnabled, pilotCertProvider, jwtPath, outputKeyCertToDir)
`
If we have such dir we need expect the root-cert exists in that directory. Now it will expects root cert exists both in outputKeyCertToDir and /var/run/secrets/istio

Reply:
OutputKeyCertToDir was originally for the location to which Istio Agent writes a key and certificate for sharing with a service. 
It seems that OutputKeyCertToDir is later extended for VM certificates. Under the PR description section, can you provide a description of the workflows related to this PR? It will help us understand the PR.